cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -44,6 +44,7 @@ var cloudproviders = []string{
 	"aws",
 	"azure",
 	"cloudstack",
+	"external",
```

Comment:
Does kubeadm use TLS bootstrapping for the kubelet serving certificates? I don't think the external cloud provider works with that feature yet (since the kubelet doesn't know what its reported addresses will be at startup)